odl/util/testutils.py

```diff
@@ -249,32 +249,102 @@ def _pass(function):
+
+    For product spaces the method is called recursively for all sub-spaces.
+
+    Notes
+    -----
+    This method is intended for internal testing purposes, for more explicit
+    example elements see `phantoms` and `LinearSpaceVector.examples`.
+
+    See Also
+    --------
+    noise_element
+    noise_vectors
+    LinearSpaceVector.examples : Examples of vectors typical to the space.
+    """
+    from odl.space import ProductSpace
+    if isinstance(space, ProductSpace):
+        return np.array([noise_array(si) for si in space])
     else:
-        arr = np.random.randn(space.size) + 1j * np.random.randn(space.size)
+        # Generate numpy vectors, real or complex or int
+        if np.issubdtype(space.dtype, np.floating):
+            arr = np.random.randn(space.size)
+        elif np.issubdtype(space.dtype, np.integer):
+            arr = np.random.randint(-10, 10, space.size)
+        else:
+            arr = (np.random.randn(space.size) +
+                   1j * np.random.randn(space.size)) / np.sqrt(2.0)
 
-    return arr.astype(space.dtype, copy=False)
+        return arr.astype(space.dtype, copy=False)
 
 
-def example_element(space):
-    return space.element(example_array(space))
+def noise_element(space):
+    """Creata a example element in ``space``
+
+    The arrays contain white noise with standard deviation 1 in the case of
+    floating point dtypes and uniformly spaced values between -10 and 10 in
+    the case of integer dtypes.
+
+    For product spaces the method is called recursively for all sub-spaces.
+
+    Notes
+    -----
+    This method is intended for internal testing purposes, for more explicit
+    example elements see `phantoms` and `LinearSpaceVector.examples`.
+
+    See Also
+    --------
+    noise_array
+    noise_vectors
+    LinearSpaceVector.examples : Examples of vectors typical to the space.
+    """
+    return space.element(noise_array(space))
 
 
-def example_vectors(space, n=1):
+def noise_vectors(space, n=1):
     """Create a list of ``n`` arrays and vectors in ``space``.
 
-    First arrays, then vectors.
+    The arrays contain white noise with standard deviation 1 in the case of
+    floating point dtypes and uniformly spaced values between -10 and 10 in
+    the case of integer dtypes.
+
+    For product spaces the method is called recursively for all sub-spaces.
+
+    Notes
+    -----
+    This method is intended for internal testing purposes, for more explicit
+    example elements see `phantoms` and `LinearSpaceVector.examples`.
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        The space to create vectors for
+    n : int
+        The number of vectors to create
+
+    Returns
+    -------
+    arrays : `numpy.ndarray`
+        A single array if ``n == 1``, otherwise a tuple of arrays.
+    arrays : ``space`` element
+        A single vector if ``n == 1``, otherwise a tuple of vector.
+
+    See Also
+    --------
+    noise_array
+    noise_element
```

Comment:
Also: return types are different for `n=1` and `n>1`. Should be the same, list or tuple.